add README.md for contrib directory

mcprentiss opened the pull request:
This is not important but having a README.md in the ~~config~~ **contrib** directory  would be helpful.
There is a lot of good stuff in here but it is a bit hard to find.
The formatting can be improved and made more beautiful, but this is a start.

mcprentiss committed:
add README.md for config directory

jiahao commented:
+1

Perhaps a table?

mcprentiss commented:
sure. something like...

Julia Config directory README  
=============================  

| Name                           | Description                                                 |  
| ------------------------------ | ---------------------------------------------------------   |
| ackrc                          | config for Ack                                              |  
| BBEditTextWrangler-julia.plist | language module for BBEdit and TextWrangler text editors    |  
| build_executable.jl            | Support building standalone executables                     |  
|build_sysimg.jl|Build a system image binary|  
|check-whitespace.sh|Check for trailing white space|  
|ctags|Add more keywords to ctags regex|  
|debug_bootstrap.gdb|bootstrap process using the debug build|  
|filterArgs.sh|Update library search code to use only tokens that start with "-L"|

mcprentiss commented:
sure something like...

Julia Config directory README  
=============================  

| Name                           | Description                                                 |  
| ------------------------------ | ---------------------------------------------------------   |
| ackrc                          | config for Ack                                              |  
| BBEditTextWrangler-julia.plist | language module for BBEdit and TextWrangler text editors    |  
| build_executable.jl            | Support building standalone executables                     |  
|build_sysimg.jl|Build a system image binary|  
|check-whitespace.sh|Check for trailing white space|  
|ctags|Add more keywords to ctags regex|  
|debug_bootstrap.gdb|bootstrap process using the debug build|  
|filterArgs.sh|Update library search code to use only tokens that start with "-L"|  
|filterDupLibs.sh|script to filter out duplicate libraries|  
|fixup-libgfortran.sh|Include libgfortran and libquadmath for installations|

ViralBShah commented:
How about separate out sections, so that the editor files are in a section by themselves?

mcprentiss commented:
okay,  they divided pretty cleanly between installation things and editors/debugging tools

Installation  
=============================  

| Name                           |  Description                                                | 
|  ----------------------------  |  ---------------------------------------------------------  |
| build_executable.jl            | Support building standalone executables                     |  
|build_sysimg.jl|Build a system image binary|  
|check-whitespace.sh|Check for trailing white space| 
|filterDupLibs.sh|script to filter out duplicate libraries|  
|fixup-libgfortran.sh|Include libgfortran and libquadmath for installations|  
|install.sh|installation script with different permissions|  
|julia.appdata.xml|Appdata config file|  
|julia.desktop|GNOME desktop config file| 
|julia.svg|julia svg image file|                                                                       
|mac/|Mac install files|  
|relative_path.sh|convert absolute path into relative path script|  
|stringreplace.c|Cygwin helper script|  
|vagrant/|portable development environment Vagrant install files|  
|windows/|windows install files|

Editors  + Debuggers
=============================  

| Name                           |  Description                                                | 
| ------------- | ------------- |
| ackrc                          | config for Ack search tool                                  | 
| BBEditTextWrangler-julia.plist | language module for BBEdit and TextWrangler text editors    | 
|ctags|Add more keywords to ctags regex search tool |  
|debug_bootstrap.gdb|bootstrap process using the debug build|  
|filterArgs.sh|Update library search code to use only tokens that start with "-L"|  
|install-tmbundle|TextMate bundle install script|  
|julia.hrc|Julia syntax highlighting definition for Colorer|  
|julia.lang|GtkSourceView (GTK+ framework) for multiline text editing config file|  
|julia-mode.el|Emacs Julia mode config file|  
|Julia_Notepad++.xml|Notepad++ config file|  
|Julia.tmbundle/|TextMate support files|  
|julia.xml|KDE editors Kate and Kwrite config file|     
|Notepad++_2_Julia.ahk|Auto Hotkey for Notepad++ config file|  
|README.ackrc.txt|README for ackrc config file|  
|README.ctags.txt|README for ctags config file|  
|relative_path.sh|convert absolute path into relative path script|  
|repackage_system_suitesparse4.make|reinstall_suitesparse4 script|    
|valgrind-julia.supp|support for Valgrind debugging tool|

ivarne commented:
+1 for that sectioned table. Can you manage clickable links for download

Editors  + Debuggers
=============================  

| Name                           |  Description                                                | 
| ------------- | ------------- |
| [ackrc](https://github.com/JuliaLang/julia/blob/master/contrib/ackrc) ([raw](https://raw.githubusercontent.com/JuliaLang/julia/master/contrib/ackrc))                          | config for Ack search tool                                  | 
| [BBEditTextWrangler-julia.plist](https://github.com/JuliaLang/julia/blob/master/contrib/BBEditTextWrangler-julia.plist) ([raw](https://raw.githubusercontent.com/JuliaLang/julia/master/contrib/BBEditTextWrangler-julia.plist)) | language module for BBEdit and TextWrangler text editors    | 
|[ctags](https://raw.githubusercontent.com/JuliaLang/julia/master/contrib/ctags) ([raw](https://raw.githubusercontent.com/JuliaLang/julia/master/contrib/ctags))|Add more keywords to ctags regex search tool |  

I'm not sure the `(raw)` links is worth the typing effort. Github features them pretty well on the main page. If you use a script to generate the readme, that would be great to include as well.

nalimilan commented:
Just a note about the name of the commit: the directory is called `contrib`, not `config`. I was really confused about the title of this PR. :-)

mcprentiss commented:
@ivarne @nalimilan Thanks I was doing this during the Super Bowl, so I am pleased there aren't more errors.  Adding links, etc would hurt the readability of the text file. Should I not be worried about it?

mcprentiss commented:
Here goes.  I think this is a converged version.

Installation
=============================

| Name                           |  Description                                                |
|  ----------------------------  |  ---------------------------------------------------------  |
|[ build_executable.jl ](https://github.com/JuliaLang/julia/blob/master/contrib/build_executable.jl) | Support building standalone executables |
|[ build_sysimg.jl ](https://github.com/JuliaLang/julia/blob/master/contrib/build_sysimg.jl) | Build a system image binary |
|[ check-whitespace.sh ](https://github.com/JuliaLang/julia/blob/master/contrib/check-whitespace.sh) | Check for trailing white space |
|[ filterDupLibs.sh ](https://github.com/JuliaLang/julia/blob/master/contrib/filterDupLibs.sh) | script to filter out duplicate libraries |
|[ fixup-libgfortran.sh ](https://github.com/JuliaLang/julia/blob/master/contrib/fixup-libgfortran.sh) | Include libgfortran and libquadmath for installations |
|[ install.sh ](https://github.com/JuliaLang/julia/blob/master/contrib/install.sh) | installation script with different permissions |
|[ julia.appdata.xml ](https://github.com/JuliaLang/julia/blob/master/contrib/julia.appdata.xml) | Appdata config file |
|[ julia.desktop ](https://github.com/JuliaLang/julia/blob/master/contrib/julia.desktop) | GNOME desktop config file |
|[ julia.svg ](https://github.com/JuliaLang/julia/blob/master/contrib/julia.svg) | julia svg image file |
|[ mac/ ](https://github.com/JuliaLang/julia/blob/master/contrib/mac/) | Mac install files |
|[ relative_path.sh ](https://github.com/JuliaLang/julia/blob/master/contrib/relative_path.sh) | convert absolute path into relative path script |
|[ stringreplace.c ](https://github.com/JuliaLang/julia/blob/master/contrib/stringreplace.c) | Cygwin helper script |
|[ vagrant/ ](https://github.com/JuliaLang/julia/blob/master/contrib/vagrant/) | portable development environment Vagrant install files |
|[ windows/ ](https://github.com/JuliaLang/julia/blob/master/contrib/windows/) | windows install files |

Editors  + Debuggers
=============================

| Name                           |  Description                                                |
| ------------------------------ | ----------------------------------------------------------- |
|[ ackrc  ](https://github.com/JuliaLang/julia/blob/master/contrib/ackrc ) |  config for Ack search tool |
|[ BBEditTextWrangler-julia.plist  ](https://github.com/JuliaLang/julia/blob/master/contrib/BBEditTextWrangler-julia.plist ) |  language module for BBEdit and TextWrangler text editors |
|[ ctags ](https://github.com/JuliaLang/julia/blob/master/contrib/ctags) | Add more keywords to ctags regex search tool |
|[ debug_bootstrap.gdb ](https://github.com/JuliaLang/julia/blob/master/contrib/debug_bootstrap.gdb) | bootstrap process using the debug build |
|[ filterArgs.sh ](https://github.com/JuliaLang/julia/blob/master/contrib/filterArgs.sh) | Update library search code to use only tokens that start with -L |
|[ install-tmbundle ](https://github.com/JuliaLang/julia/blob/master/contrib/install-tmbundle) | TextMate bundle install script |
|[ julia.hrc ](https://github.com/JuliaLang/julia/blob/master/contrib/julia.hrc) | Julia syntax highlighting definition for Colorer |
|[ julia.lang ](https://github.com/JuliaLang/julia/blob/master/contrib/julia.lang) | GtkSourceView (GTK+ framework) for multiline text editing config file |
|[ julia-mode.el ](https://github.com/JuliaLang/julia/blob/master/contrib/julia-mode.el) | Emacs Julia mode config file |
|[ Julia_Notepad++.xml ](https://github.com/JuliaLang/julia/blob/master/contrib/Julia_Notepad++.xml) | Notepad++ config file |
|[ Julia.tmbundle/ ](https://github.com/JuliaLang/julia/blob/master/contrib/Julia.tmbundle/) | TextMate support files |
|[ julia.xml ](https://github.com/JuliaLang/julia/blob/master/contrib/julia.xml) | KDE editors Kate and Kwrite config file |
|[ Notepad++_2_Julia.ahk ](https://github.com/JuliaLang/julia/blob/master/contrib/Notepad++_2_Julia.ahk) | Auto Hotkey for Notepad++ config file |
|[ README.ackrc.txt ](https://github.com/JuliaLang/julia/blob/master/contrib/README.ackrc.txt) | README for ackrc config file |
|[ README.ctags.txt ](https://github.com/JuliaLang/julia/blob/master/contrib/README.ctags.txt) | README for ctags config file |
|[ repackage_system_suitesparse4.make ](https://github.com/JuliaLang/julia/blob/master/contrib/repackage_system_suitesparse4.make) | reinstall_suitesparse4 script |
|[ valgrind-julia.supp ](https://github.com/JuliaLang/julia/blob/master/contrib/valgrind-julia.supp) | support for Valgrind debugging tool |

tkelman commented on contrib/README.md:
```diff
@@ -0,0 +1,33 @@
+README.ctags.txt - README for ctags config file  
+relative_path.sh - convert absolute path into relative path script  
+repackage_system_suitesparse4.make - reinstall_suitesparse4 script  
+stringreplace.c - Cygwin helper script  
```
This isn't for Cygwin, we use it everywhere during `make install`. It's to replace strings to hardcoded paths in binaries. It just has a limited history with the current file name, it used to be called stringpatch but Windows does funny things with admin rights and 32-bit executables that have "patch" or "install" in their names. Here's the history of the file at its old name: https://github.com/JuliaLang/julia/commits/b42d5807cfbae15e520dcf23c3eff32addb3e437/contrib/stringpatch.c

mcprentiss committed:
new updates to README.md in /contrib

Added links, added fixup-libstdc++.sh, separated into INSTALLATION, and EDITOR and DEBUGGING sections, fixed description of stringreplace.c. I think this is read to merge.

mcprentiss commented:
@jiahao @ViralBShah @tkelman I think this is ready to go. I made the required fixes.

ivarne commented:
Thanks for doing this!
